seqr/views/utils/orm_to_json_utils.py

```diff
@@ -757,10 +757,21 @@ def get_json_for_matchmaker_submission(submission):
         additional_model_fields=['contact_name', 'contact_href', 'submission_id'])
 
 
-def get_json_for_rna_seq_outliers(models, **kwargs):
-    additional_values = {
-        'isSignificant': Case(
-            When(p_adjust__lt=models.model.SIGNIFICANCE_THRESHOLD, then=Value(True)), default=Value(False)
-        ),
-    }
-    return get_json_for_queryset(models, additional_values=additional_values, **kwargs)
+MAX_SIGNIFICANT_OUTLIER_NUM = 50
+
+
+def get_json_for_rna_seq_outliers(models, top_outliers_only=False, nested_fields=None, **kwargs):
```

Comment:
I agree that populating two different data to a single redux state field can be risky. But the front end should have done enough checks if it is required to reload the data.

About moving the grouping by gene into this function, why don't we also generate the whole `rnaSeqData` response in the shared function so that we don't have to generate it in different places?

It is okay to make changes to the `nested_fields`, `kwargs`, and others.

Reply:
>  why don't we also generate the whole rnaSeqData response in the shared function so that we don't have to generate it in different places?

I like this idea!